Library/Homebrew/software_spec.rb

```diff
@@ -513,7 +520,30 @@ def compatible_locations?(tag: Utils::Bottles.tag)
       tag.default_cellar
     end
 
-    return true if [:any, :any_skip_relocation].include?(cellar)
+    return true if RELOCATABLE_CELLARS.include?(cellar)
+
+    prefix = Pathname(cellar).parent.to_s
+
+    cellar_relocatable = cellar.size >= HOMEBREW_CELLAR.to_s.size && ENV["HOMEBREW_RELOCATE_BUILD_PREFIX"]
+    prefix_relocatable = prefix.size >= HOMEBREW_PREFIX.to_s.size && ENV["HOMEBREW_RELOCATE_BUILD_PREFIX"]
+
+    compatible_cellar = cellar == HOMEBREW_CELLAR.to_s || cellar_relocatable
+    compatible_prefix = prefix == HOMEBREW_PREFIX.to_s || prefix_relocatable
+
+    compatible_cellar && compatible_prefix
+  end
+
+  # Should the build prefix for the {Bottle} this {BottleSpecification} belongs to be relocated?
```

Comment:
Phrasing the docstring as a question here is a little weird.

Reply:
Agreed.